### PR TITLE
Add USERAGENT constant

### PR DESCRIPTION
**Pull request by buthed010203**

### CODE REVIEW

File: src/constants.ts [hunk 1]
```diff
@@ -8,6 +10,8 @@ export const EXTENSIONS = ["mkv", "mp4", "avi"];
 
 export const TORRENT_CACHE_FOLDER = "torrent_cache";
 
+export const USERAGENT = `CrossSeed/${packageDotJson.version}`;
```
Comment (mmgoodnow) [inline src/constants.ts:13]:
```suggestion
export const USER_AGENT = `CrossSeed/${packageDotJson.version}`;
```

File: src/torrent.ts [hunk 2]
```diff
@@ -11,6 +11,7 @@ import { getRuntimeConfig, NonceOptions } from "./runtimeConfig.js";
 import { createSearcheeFromTorrentFile, Searchee } from "./searchee.js";
 import { stripExtension } from "./utils.js";
 import fetch, { Response } from "node-fetch";
+import { USERAGENT } from "./constants.ts";
```
Comment (mmgoodnow) [inline src/torrent.ts:14]:
```suggestion
import { USER_AGENT } from "./constants.ts";
```

File: src/torrent.ts [hunk 3]
```diff
@@ -41,7 +42,7 @@ export async function parseTorrentFromURL(
 	let response: Response;
 	try {
 		response = await fetch(url, {
-			headers: { "User-Agent": "cross-seed" },
+			headers: { "User-Agent": USERAGENT },
```
Comment (mmgoodnow) [inline src/torrent.ts:45]:
```suggestion
			headers: { "User-Agent": USER_AGENT },
```

File: src/torznab.ts [hunk 4]
```diff
@@ -21,6 +21,7 @@ import {
 	reformatTitleForSearching,
 	stripExtension,
 } from "./utils.js";
+import { USERAGENT } from "./constants.ts";
```
Comment (mmgoodnow) [inline src/torznab.ts:24]:
```suggestion
import { USER_AGENT } from "./constants.ts";
```

File: src/torznab.ts [hunk 5]
```diff
@@ -420,7 +421,7 @@ async function makeRequests(
 	const outcomes = await Promise.allSettled<Candidate[]>(
 		searchUrls.map((url, i) =>
 			fetch(url, {
-				headers: { "User-Agent": "cross-seed" },
+				headers: { "User-Agent": USERAGENT },
```
Comment (mmgoodnow) [inline src/torznab.ts:424]:
```suggestion
				headers: { "User-Agent": USER_AGENT },
```

File: src/constants.ts [hunk 6]
```diff
@@ -1,3 +1,5 @@
+const packageDotJson = require("../package.json");
```
Comment (mmgoodnow) [inline src/constants.ts:1]:
```suggestion
import { createRequire } from "module";
const require = createRequire(import.meta.url);
const packageDotJson = require("../package.json");
```

**Comment by buthed010203** [comment]
@mmgoodnow Any eta on this?

**Comment by mmgoodnow** [comment]
waiting for things to settle down after the big 5.0 release. probably another couple weeks

**Comment by buthed010203** [comment]
Sounds good, thanks for letting me know

**Comment by mmgoodnow** [comment]
v5.2.0

**Comment by buthed010203** [comment]
Cheers!

**Pull request merged**